### ISSUE TITLE
Fix Object.assign example

### ISSUE THREAD

**Issue by connec** [ryanmcdermott/clean-code-javascript#4]
The `Object.assign` example is currently incorrect.

```js
function createMenu(config) {
  Object.assign(config, {
    title: 'Foo',
    body: 'Bar',
    buttonText: 'Baz',
    cancellable: true
  });
}

createMenu({ title: 'Not Foo' });
```

The `title` property will always be set to `'Foo'`!

I was going to make a PR with a fix, but the approach really depends on whether the mutation of `config` is desirable.

How I would do it (this does not behave identically to the "bad" example):

```js
function createMenu(config) {
  // makes a copy of `config`, with default values
  config = Object.assign({
    title: 'Foo',
    body: 'Bar',
    buttonText: 'Baz',
    cancellable: true
  }, config);
}
```

How you could do it with mutation (this is identical to the "bad" example):

```js
function createMenu(config) {
  // mutates `config`, setting default values
  Object.assign(config, Object.assign({
    title: 'Foo',
    body: 'Bar',
    buttonText: 'Baz',
    cancellable: true
  }, config));
}
```

**Comment by halfzebra** [comment]
Sorry if I'm overthinking it, but I find it rather confusing to provide the example of object mutation in a procedural style, while advising to [favor functional programming over imperative programming.](https://github.com/ryanmcdermott/clean-code-javascript#favor-functional-programming-over-imperative-programming)

**Comment by rkichenama** [comment]
If the purpose is to mutate `config` to add defaults but keep changes, then the example should be
```javascript
function createMenu(config) {
  Object.assign(
    config, // object to merge changes into
    {
      title: 'Foo',
      body: 'Bar',
      buttonText: 'Baz',
      cancellable: true
    }, // defaults
    config // overrides
);
}

createMenu({ title: 'Not Foo' });
```

IMHO, I think this is the case since there is no return and we are dealing with objects, passed by reference.

**Comment by kwelch** [comment]
I would say the best option would be to maintain the passed in parameter and create a new object all together that use defaults then user param. 

```
function createMenu(options) {
  let config = Object.assign({}, {
      title: 'Foo',
      body: 'Bar',
      buttonText: 'Baz',
      cancellable: true
    }, // defaults
    options // overrides
  );
}
```

**Comment by rkichenama** [comment]
Good have to return the be object `config`

**Comment by connec** [comment]
@rkichenama that example does not work because `config` has already mutated when the 'overrides' are applied:

```js
let config = { key: 'override' }
Object.assign(config, { key: 'default' }, config)
console.log(config.key) // default
```

Hence the need for an additional `Object.assign`.

@kwelch that option is equivalent to the non-mutating version I specified - the defaults are already in a new object, so merging them into *another* new object is redundant.

---

Anyway, looks like this was fixed in https://github.com/ryanmcdermott/clean-code-javascript/commit/d00ca4c70c97076b9678c193b125fdf9b810b89e with @kwelch's version.
